pkg/volume/glusterfs/glusterfs.go

```diff
@@ -145,6 +145,19 @@ func (plugin *glusterfsPlugin) execCommand(command string, args []string) ([]byt
 	return cmd.CombinedOutput()
 }
 
+func (plugin *glusterfsPlugin) ConstructVolumeSpec(volumeName, mountPath string) (*volume.Spec, error) {
+	glusterfsVolume := &api.Volume{
+		Name: volumeName,
+		VolumeSource: api.VolumeSource{
+			Glusterfs: &api.GlusterfsVolumeSource{
+				EndpointsName: volumeName,
+				Path:          volumeName,
```

Comment:
These fields such as EndpointsName and Path was used for constructing UniqueVolumeName (returned by plugin.GetVolumeName()). If leaving it unset, the returned GetVolumeName() will be empty and the returned GetUniqueVolumeName() will only has pluginName which will cause name collision. But after Paul's change, for nonattachable volume, these device fields are not used for constructing UniquevVolumeName, so leaving them unset seems ok. My only question is in api.types comments, it saids some of these fields are requirement, is it ok to leave them unset?